lars-reimann/api-editor
# feat: filter modules, classes, functions, and parameters

Closes #7.

## Summary of Changes
* Filter box in the menu bar.
* Expected input has the form `<scope>:<filterString>`, where `scope` is one of module, class, function, or parameter and `filterString` is an arbitrary string. Each scope must occur at most once.
* The filterString must occur anywhere in the name of the declaration for it to be matched. This means `module:neighbors` would match `sklearn.neighbors._kde`, for example.
* Search is case-insensitive.
* When the class filter is active, all top-level functions are removed.
* When any filter is active, nodes in the tree view without children are removed.

## Screenshots
![image](https://user-images.githubusercontent.com/2501322/126671545-85f068ad-15e0-4f17-9d7e-1f44c1eac4f5.png)

![image](https://user-images.githubusercontent.com/2501322/126671574-b0684626-4549-433d-a4ca-bc1869cb9ec8.png)

## Testing instructions
* Try the following queries and vary them:
  * module:neighbors
  * class:transformer
  * function:fit
  * parameter:y
* Check that the original view is reinstantiated when the filters are removed
